### PR TITLE
devmode: make reusable

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
the `devmode` helper made for the Nixpkgs/NixOS manual was exposed wrapped
in `mkShell`, which made it impossible to reuse.

this change strips that wrapper and reproduces it at the call site.

now one can use `devmode` from anywhere Nixpkgs is available:

    devmode = pkgs.devmode.override {
      buildArgs = toString ./.;
      open = "/index.html";
    };

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested: ran `nix-shell --run devmode` in `doc` and `nixos/doc/manual`
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/nix/web-devmode.nix [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
+    cat << EOF
+    <!DOCTYPE html>
     <html>
     <head>
       <style>
         @media (prefers-color-scheme: dark) {
           :root { filter: invert(100%); }
         }
-      </style>
-    </head>
```
Comment (GetPsyched) [inline pkgs/tools/nix/web-devmode.nix:27]: Seems accidental?

Comment (fricklerhandwerk) [reply pkgs/tools/nix/web-devmode.nix]: Yes, thanks for catching that.

File: doc/doc-support/package.nix [hunk 2]
```diff
@@ -95,10 +97,14 @@ stdenvNoCC.mkDerivation (
...
-        open = "/share/doc/nixpkgs/manual.html";
-      };
+      shell =
+        let
+          devmode' = devmode.override {
+            buildArgs = "./.";
+            open = "/share/doc/nixpkgs/manual.html";
+          };
+        in
+        mkShellNoCC { packages = [ devmode' ]; };
```
Comment (mightyiam) [inline doc/doc-support/package.nix:107]: Good to learn of `mkShellNoCC`

File: pkgs/by-name/de/devmode/package.nix [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
-  open,
+  open ? "/index.html",
 }:
 let
   inherit (nodejs_latest.pkgs) live-server;
 
-  error_page = writeShellScriptBin "error_page" ''
-    echo "<!DOCTYPE html>
+  error-page = writeShellScriptBin "error-page" ''
+    cat << EOF
```
Comment (mightyiam) [inline pkgs/by-name/de/devmode/package.nix:18]: Neat trick

**Review by mightyiam — APPROVED** [review]
If it has been tested, I'm good.

**Pull request merged**